### PR TITLE
(java-sdk) Add tests for Manifest and polish MemoryOptions

### PR DESCRIPTION
**Pull request by thomasdarimont**
- Fix broken javadoc comment
- Add tests for Manifest serilization and refactor tests
- Remove duplicate developer entry in pom.xml
- Add initial readme

Improves #117

### CODE REVIEW

File: java/src/main/java/org/extism/sdk/manifest/MemoryOptions.java [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-        Integer max // Integer to allow `null` as max
-//
-) {
+/**
+ * Configures memory for the Wasm runtime.
+ * Memory is described in units of pages (64KB) and represent contiguous chunks of addressable memory.
+ *
+ * @param maxPages Max number of pages.
+ */
+public record MemoryOptions(@SerializedName("max_pages") Integer maxPages) {
```
Comment (bhelx) [inline java/src/main/java/org/extism/sdk/manifest/MemoryOptions.java:11]: @nilslice we decided to hold off on this change right? Just making sure.

Comment (nilslice) [reply java/src/main/java/org/extism/sdk/manifest/MemoryOptions.java:11]: Yea, I think we will ship this as-is so it's immediately usable with the current release, and then we can make SDK-wide changes for the next release.

Comment (nilslice) [reply java/src/main/java/org/extism/sdk/manifest/MemoryOptions.java:11]: by as-is, I mean using `"max"` as the name :)

Comment (bhelx) [reply java/src/main/java/org/extism/sdk/manifest/MemoryOptions.java:11]: @thomasdarimont if you don't mind let's revert the change back to max for the time being.

Comment (thomasdarimont) [reply java/src/main/java/org/extism/sdk/manifest/MemoryOptions.java]: I renamed the `MemoryOptions` field back to `max`.

**Pull request merged**